src/Action/Admin/OrderRefundsListAction.php

```diff
@@ -35,14 +39,32 @@ public function __construct(
         private Environment $twig,
         private RequestStack $requestStack,
         private UrlGeneratorInterface $router,
+        private ?StateMachineInterface $stateMachine = null,
```

Comment:
Maybe create a checker service that would keep logic inside and use that instead?
Same in twig, a simple function would suffice instead of hardcoding particular transitions 🤷